bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpRequestBuilder.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Builder class to construct http requests
+ *
+ * @author Martin van Wingerden - initial API
+ */
+public class HttpRequestBuilder {
+    private static final int DEFAULT_TIMEOUT_IN_MS = 5_000;
+
+    private final String method;
+    private final String url;
+    private InputStream inputStream;
+    private int timeoutMilliseconds = DEFAULT_TIMEOUT_IN_MS;
+    private Properties headers;
+    private String contentType;
+
+    /**
+     * Private constructor, to hide away the construction behind some factory methods
+     *
+     * @param method Http method to be used
+     * @param url url to fetch or post data to
+     */
+    private HttpRequestBuilder(String method, String url) {
```

Comment:
The invariants:

```java
import static java.util.Objects.requireNonNull;
....
....
requireNonNull(method, "HTTP Method cannot be null");
requireNonNull(url, "URL cannot be null");
```

If you use JDT annotations, you would need to annotate the static methods.

Reply:
I added the annotations